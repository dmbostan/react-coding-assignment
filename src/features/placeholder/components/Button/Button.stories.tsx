import type { Meta, StoryObj } from '@storybook/react';
import Button from './Button';

const meta: Meta<typeof Button> = {
  component: Button,
};

export default meta;

type Story = StoryObj<typeof Button>;

const clickHandler = () => console.log('Clicked.');

export const Default: Story = {
  render: () => <Button clickHandler={() => clickHandler()}>Click me</Button>,
};
