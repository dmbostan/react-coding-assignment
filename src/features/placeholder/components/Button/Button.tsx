import { ButtonTypes } from './Button.types.ts';
import styles from './Button.module.css';
import classNamesHandler from '@shared/utils/classNamesHandler.ts';

const Button = ({ clickHandler, className, children }: ButtonTypes) => {
  const classNames = classNamesHandler(className, styles.button);

  return (
    <button onClick={clickHandler} className={classNames}>
      {children}
    </button>
  );
};

export default Button;
