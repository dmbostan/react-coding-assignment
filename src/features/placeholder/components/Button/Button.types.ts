import { ReactNode } from 'react';

export type ButtonTypes = {
  clickHandler: () => void;
  className?: string;
  children: string | ReactNode | ReactNode[] | null;
};
