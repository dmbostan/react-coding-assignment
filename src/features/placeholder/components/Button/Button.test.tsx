import { render, fireEvent } from '@testing-library/react';
import Button from './Button.tsx';

describe('Button component', () => {
  it('should render a button with the provided text and class name', () => {
    const clickHandler = vi.fn();
    const className = 'custom-button';
    const buttonText = 'Click me';

    const { getByText } = render(
      <Button clickHandler={clickHandler} className={className}>
        {buttonText}
      </Button>,
    );

    const buttonElement = getByText(buttonText);

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass(className);
  });

  it('should call the click handler when clicked', () => {
    const clickHandler = vi.fn();
    const className = 'custom-button';
    const buttonText = 'Click me';

    const { getByText } = render(
      <Button clickHandler={clickHandler} className={className}>
        {buttonText}
      </Button>,
    );

    const buttonElement = getByText(buttonText);
    fireEvent.click(buttonElement);

    expect(clickHandler).toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    const clickHandler = vi.fn();
    const className = 'custom-button';
    const buttonText = 'Click me';

    const { asFragment } = render(
      <Button clickHandler={clickHandler} className={className}>
        {buttonText}
      </Button>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
