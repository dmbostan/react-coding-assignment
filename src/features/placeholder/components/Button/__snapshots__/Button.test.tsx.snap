// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="custom-button _button_b44ff9"
  >
    Click me
  </button>
</DocumentFragment>
`;
